src/app/_components/Specialists/DistrictFilter.js

```diff
@@ -0,0 +1,78 @@
+'use client';
+
+import { useState } from 'react';
+import { CircularProgress } from '@mui/material';
+import { useListDistrict, useSetParam } from '@hooks';
+import { CheckBox } from '@components/CheckBox';
+import { ClearFilterButton, FilterBase } from '@components/Specialists';
+import { useSearchParams } from 'next/navigation';
+import PropTypes from 'prop-types';
+
+function DistrictList({ setCount, defaultValue }) {
+  const { data: districts, isLoading } = useListDistrict();
+  const [selectedDistricts, setSelectedDistricts] = useState(defaultValue);
+
+  const { addParam, deleteParam } = useSetParam('district');
+
+  const onChange = district => {
+    if (selectedDistricts.includes(district)) {
+      setSelectedDistricts(selectedDistricts.filter(selDistrict => selDistrict !== district));
+      setCount(count => count - 1);
+      deleteParam(district);
+    } else {
+      setSelectedDistricts([...selectedDistricts, district]);
+      setCount(count => count + 1);
+      addParam(district);
+    }
+  };
+
+  if (isLoading) return <CircularProgress />;
+
+  if (!isLoading && !districts) return null;
+
+  if (districts) {
```

Comment:
@watchlar2000 @alex-popov-tech It should  Dont do something?.length && ..., it will render 0 if the length is 0. So either add ternary operator and return null or transform districts?.length to boolean

Reply:
Guys, please refer to this PR [MID-69](https://trello.com/c/fViJdFP2/69-%D1%8E%D0%B7%D0%B5%D1%80-%D1%80%D0%BE%D0%B7%D0%B4%D1%96%D0%BB-%D1%81%D0%BF%D0%B5%D1%86%D1%96%D0%B0%D0%BB%D1%96%D1%81%D1%82%D0%B8-fe)

Reply:
it's related to this pr https://github.com/keenethics/mayak/pull/71